### PR TITLE
feat(header): add profile.description prop to the popover

### LINKED ISSUE #668: Ability to customize the dropdown when clicking on a header / top button.
## Request -
Our team currently needs the ability to customize what is rendered when a user clicks the User Profile button to activate the dropdown.
Currently clicking the button activates a dropdown with some information.
However we would like to add additional information to that dropdown.

- We require an email address below the First name Last name. (I see there is currently a span rendered for this very purpose)
- We also require some additional links and or verbiage on the `body` of the dropdown.

### Details

- **Application / team -** 
Cloud Identity
- **Work item -** 

- **Needed by -** 
We were hoping to have this by Friday. But I do not want to cause undue stress in making this feature available.

### Proposal

**For our immediate needs, we would like to have a `security--header__popover__body` section which can render custom JSX elements. If we could additionally insert the email address below the user name. That would be sufficient for our needs.**

My initial thought is that the onClick for the Profile button would be able to call a method which returns a JSX element.
This method => JSX.Element would override what currently shows when a user clicks the User Profile button.

Its possible this could be extended for the notifications and helps icon to add a custom render method as well.

Or get really deep and update the `renderAddons` method with a property to select `position` with values of `sidenav | topnav`. This would extend the functionality of renderAddons so that in the similar manner in which it currently adds a button and action to the leftnav... You could add a button to the topnav which renders a customizable dropdown

Please ignore the lack of styling in the dropdown body. 

image

`<span class='security--transition'>
	<div class='security--header__popover'>
		<div value='profile' tabindex='0' role='tabpanel' class='security--header__popover--focus'>
			<section class='security--header__popover__header security--header__popover__profile__header'>
				<span class='security--profile-image security--header__popover__profile__header__icon security--profile-image--large'>
					SU
				</span>
				<div>
					<span class='security--header__popover__profile__header__title'>Sample User</span>
					{/* 
						This span already exists. Possibly add a property to the Profile object which adds the email address and automatically links it?
						Would probably require a className so that the link can be unstyled if needed.
					*/}
					<span class='security--header__popover__profile__header__email'>test-user@email.com</span>
				</div>
			</section>
			{/* 
				Can we have a Popover BODY section? There is already a Header and a Footer.
				This is where we would like to render custom JSX
			*/}
			<section class='security--header__popover__body'>
				<a href='#'>Additional link</a>
				<p>This may be additional verbiage also added in the body area of the dropdown</p>
			</section>
			<section class='security--header__popover__footer'>
				<a
					href='#'
					class='bx--link security--header__popover__link security--header__popover__link--secondary security--header__popover__profile__footer__edit'
				>
					Edit profile
				</a>
				<a
					href='#'
					class='bx--link security--header__popover__link security--header__popover__link--secondary security--header__popover__profile__footer__logout'
				>
					Sign out
				</a>
			</section>
		</div>
	</div>
</span>
`

### Goals / use cases

- [ ] Provide additional guidance and or links in the body of User Profile dropdown.
- [x] Add an email address below the user name in the Profile dropdown

### PR DESCRIPTION
**Pull request by jendowns**
## Affected issues

- Resolves #668

I have opted to add the extra section below the profile popover heading, and above the body section. But let me know if it needs to be moved.

## Proposed changes

- adds `profile.description` prop with `node` type
- update stories to include the prop in use
- update snapshots

### CODE REVIEW

**Comment by SimonFinney** [comment]
Hey @Acts7 ! This is ready for review at the deploy preview URL above

File: src/components/Header/_mocks_/index.js [hunk 1]
```diff
@@ -100,6 +101,12 @@ const profile = {
     first_name: 'Sample',
     surname: 'User',
   },
+  description: (
```
Comment (SimonFinney) [inline src/components/Header/_mocks_/index.js:104]: 🔥

Comment (jendowns) [reply src/components/Header/_mocks_/index.js:104]: Hey, does this mean the changes are approved?

**Comment by jendowns** [comment]
Hey @petervachon as far as I know, I think it's meant to be the latter: an additional area to add text. (@Acts7 do you have more more?)

**Comment by jendowns** [comment]
Hey @Acts7, are you okay with these changes? Do they resolve #668 for you?

**Comment by Acts7** [comment]
@jendowns @SimonFinney 
This looks terrific. Exactly what we needed. Thank you!!

**Comment by Acts7** [comment]
@SimonFinney @jendowns THANK YOU THANK YOU!!!

**Pull request merged**